webview-ui/src/components/chat/ModelPickerModal.tsx

```diff
@@ -460,157 +592,200 @@ const ModelPickerModal: React.FC<ModelPickerModalProps> = ({ isOpen, onOpenChang
+										type="range"
+										value={thinkingEnabled ? thinkingBudget : 0}
+									/>
+								</ThinkingSliderRow>
+							)}
 						</SettingsSection>
 
 						{/* Scrollable content */}
 						<ModelListContainer>
-							{/* Provider list - shown when expanded, inside scroll area */}
-							{isProviderExpanded && (
-								<>
-									{configuredProviders.map((provider) => (
-										<ProviderListItem
-											$isSelected={provider === selectedProvider}
-											key={provider}
-											onClick={() => handleProviderSelect(provider)}>
-											<span>{getProviderLabel(provider)}</span>
-											{provider === selectedProvider && (
+							<>
+								{/* Current model - inside scroll area for seamless scrolling */}
+								{isSplit ? (
+									<SplitModeRow onClick={(e) => e.stopPropagation()}>
+										<Tooltip>
+											<TooltipTrigger asChild>
+												<SplitModeCell
+													$isActive={activeEditMode === "plan"}
+													onClick={() => setActiveEditMode("plan")}>
+													<SplitModeLabel $mode="plan">P</SplitModeLabel>
+													<SplitModeModel>
+														{planModel.selectedModelId?.split("/").pop() || "Not set"}
+													</SplitModeModel>
+												</SplitModeCell>
+											</TooltipTrigger>
+											<TooltipContent side="top" style={{ zIndex: 9999 }}>
+												Plan mode
+											</TooltipContent>
+										</Tooltip>
+										<Tooltip>
+											<TooltipTrigger asChild>
+												<SplitModeCell
+													$isActive={activeEditMode === "act"}
+													onClick={() => setActiveEditMode("act")}>
+													<SplitModeLabel $mode="act">A</SplitModeLabel>
+													<SplitModeModel>
+														{actModel.selectedModelId?.split("/").pop() || "Not set"}
+													</SplitModeModel>
+												</SplitModeCell>
+											</TooltipTrigger>
+											<TooltipContent side="top" style={{ zIndex: 9999 }}>
+												Act mode
+											</TooltipContent>
+										</Tooltip>
+									</SplitModeRow>
+								) : (
+									selectedModelId &&
+									modelBelongsToProvider &&
+									(() => {
+										// Check if current model has a featured label (only for Cline provider)
+										const currentFeaturedModel = isClineProvider
+											? [...recommendedModels, ...freeModels].find((m) => m.id === selectedModelId)
+											: undefined
+										return (
+											<CurrentModelRow onClick={() => onOpenChange(false)}>
+												<ModelInfoRow>
+													<ModelName>{selectedModelId.split("/").pop() || selectedModelId}</ModelName>
+													<ModelProvider>
+														{OPENROUTER_MODEL_PROVIDERS.includes(selectedProvider)
+															? selectedModelId.split("/")[0]
+															: selectedProvider}
+													</ModelProvider>
+												</ModelInfoRow>
+												{currentFeaturedModel?.label && (
+													<ModelLabel>{currentFeaturedModel.label}</ModelLabel>
+												)}
 												<Check
```

Comment:
nit: the check doesn't align with the label below

```suggestion
												<Check className="mr-2 size-2" />
```

Before:

<img width="812" height="301" alt="image" src="https://github.com/user-attachments/assets/c4a709af-c95c-404f-afa7-0847eb8143b9" />


After:

<img width="820" height="307" alt="image" src="https://github.com/user-attachments/assets/c8fbf7ca-f27d-4cea-8970-7af829f4469d" />